homeassistant/components/songpal/media_player.py

```diff
@@ -240,6 +270,46 @@ async def async_set_sound_setting(self, name, value):
         """Change a setting on the device."""
         _LOGGER.debug("Calling set_sound_setting with %s: %s", name, value)
         await self._dev.set_sound_settings(name, value)
+    
+    
+    async def async_create_group(self, group_name, slaves):
+        """Create a new group."""
+        
+        uuids = [
+            device._uuid
+            for device in self._hass.data[SONGPALDATA]
+            if device._entity_id in slaves
+        ]
+        _LOGGER.info("Calling create_group to create %s on %s with %s", group_name, self._name, slaves)
+        await self._group.connect()
+        await self._group.create(group_name, uuids)
+        self._is_coordinator = True
+        self._is_grouped = True ##TODO: set slaves grouped
+        self._group_list = slaves
+        for device in self._hass.data[SONGPALDATA]:
+            if device._entity_id in slaves:
+                device._is_grouped = True
```

Comment:
Please don't share the entity instance with other entities. Either build a manager for the groups, that only need to know the entity_id of the involved entities, or use our dispatch helper to communicate between entities.

Here's an example of a group manager:
https://github.com/home-assistant/core/blob/be0ef5ad6c437ece3443cf4dfda1f719b299d0f0/homeassistant/components/heos/__init__.py#L239

Reply:
Thanks @MartinHjelmare for the detailed feedback, I'll try and see if these changes are withing my abilities